File: test/data/typescript/test06/StateState1.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateState1 extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitAborted(): BaseState | null
    {
        return null;
    }
    public TransitSucceeded(): BaseState | null
    {
        this._stateController.InstanceOfState2.Setup(false, false);
        return this._stateController.InstanceOfState2;
    }
    public  GetStateName(): string
    {
        return "State1";
    }
    public GetParent(): BaseState | null
    {
        return null;
    }
}

File: test/data/cpp/test06/StateState1.cpp
#include "StateState1.h"
StateState1::StateState1(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateState1::~StateState1()
{
}
BaseState* StateState1::TransitAborted()
{
    return 0;
}
BaseState* StateState1::TransitSucceeded()
{
    _stateController->InstanceOfState2->Setup(false, false);
    return _stateController->InstanceOfState2;
}
const char* StateState1::GetStateName()
{
    return "State1";
}
BaseState* StateState1::GetParent()
{
    return 0;
}